WordPress/openverse-frontend
# Fix pot file generation

## Fixes


Fixes #317 by @sarayourfriend

## Description


This PR replaces the recursive searching for a JSON property in the `en.json` file in `findPath` function  with the code that first extracts all the full-path JSON keys ( eg. `browse-page.search-form.button`) and then for each of these keys saves the relevant information to the POT file.
The code for JSON manipulation (key extraction, and value extraction using the full-path JSON keys) was copied from https://github.com/coderaiser/all-object-keys and // https://github.com/coderaiser/jessy .

Previously, the recursive `findPath` function was finding the first available value for property specified. For example, when looking for `button`, it would find `home.search.button` and use it for any key that had `button` in it, even for `browse-page.search-form.button`. And even though recursive way of finding the property in a deeply-nested JSON was fun, it's easier to debug the current setup :) 

I had some problems resolving paths and reading a `json` file, and also in light of #313, I replaced the `js` files in `locales` folder with `mjs` files. I am not sure it is correct, and would be happy if someone could improve imports/paths.

Also, I could not get the `mjs` tests to run :(

## Testing Instructions

Try to generate the POT file using `npm run i18n:generate-pot` and observe that all values are saved to 'openverse.pot' file correctly. For example, now there exist two different objects for Search button: `browse-page.search-form.button` and `home.search.button`, instead of two copies of `home.search.button` as before.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
